compiler/src/dotty/tools/dotc/util/SourceFile.scala

```diff
@@ -197,12 +194,7 @@ class SourceFile(val file: AbstractFile, computeContent: => Array[Char]) extends
   /** The column corresponding to `offset`, starting at 0 */
   def column(offset: Int): Int = {
     var idx = startOfLine(offset)
-    var col = 0
-    while (idx != offset) {
-      col += (if (idx < content().length && content()(idx) == '\t') (tabInc - col) % tabInc else 1)
```

Comment:
Nice catch. That is a clever way to keep tabs from creeping into your code base. Confirming it was more correct in the original. There were a couple of bug fixes since ancient times, I'll check whether they need forwardly porting.

Reply:
@smarter at least I can confirm that in scala2 it doesn't affect semanticdb generation as this method isn't used in plugin.
Even without braces as in Scala2 I wouldn't say that it's defined correctly.

For example:
```scala
// scala 2.13.8
object  A {
	// indented with one tab
	def foo: Unit = ???
	def foo: Unit = ???
	      ^^^^
	      error: method foo is defined twice;
                the conflicting method foo was defined at line 6:13
}
```
Notice that position in message is `6:13` while it should be `6:7`

Reply:
Where does the example output come from? Column != position. In Scala 2, leftmost column is column 1, in Scala 3 it is zero. So `foo` is in the 13th column. In this rendering, there is also an extra caret. What is `7`?

Reply:
I assume that when a line with an error contains tabs, these things are true in Scala 2:
- The error caret is displayed at the correct position when running the scalac shell script directly
- The error caret is displayed at the correct position when running the compiler via sbt
- The error squiggly line is displayed at the correct position in Metals

Are these things also true for Scala 3 with your patch?

Reply:
the problem with tab-to-space conversion is, you either have to _know_ the width of a tab, or you have to assume one which will be wrong in many cases. i've seen scala code using 2, 3, 4 and 8 spaces wide tabs in the wild. imho the only halfway sane thing is to have character indices in the columns information, and only assume a certain width where it's necessary to render a caret in an error message.

Reply:
I see `MessageRendering.errorMsg` uses `startColumnPadding` which is tab-preserving but then uses length to determine a count of spaces.

(Sorry, I didn't realize we're talking about post-processing the tabs-translated message.)

In Scala 2, source had a notion of underlying source for scripts and any file where positions are translated.

The problem of rendered char column vs index of code point is the same for Unicode.

I agree that all this is a rendering concern. For example, the ambitious `startColumnPadding` is just the slice from `startOfLine` to the `offset`, which can be rendered any way you like.

Reply:
ok, then I guess this change is fine and I'll let others decide if more needs to be done.